main.tf

```
@@ -9,8 +9,31 @@ module "label" {
   tags       = var.tags
 }
 
+data "aws_caller_identity" "current" {}
+
 data "aws_iam_policy_document" "kms" {
   count = var.enabled ? 1 : 0
+  
+  statement {
+    sid    = "Enable IAM User Permissions"
+    effect = "Allow"
+
+    actions = [
+      "kms:*"
```

```suggestion
      "kms:Create*",
      "kms:Describe*",
      "kms:Enable*",
      "kms:List*",
      "kms:Put*",
      "kms:Update*",
      "kms:Revoke*",
      "kms:Disable*",
      "kms:Get*",
      "kms:Delete*",
      "kms:ScheduleKeyDeletion",
      "kms:CancelKeyDeletion"
```
Hi @andrewmackett, just a suggestion - I believe the [recommendation](https://aws.amazon.com/premiumsupport/knowledge-center/update-key-policy-future/) from AWS is to use specific permissions, to avoid giving unnecessary decrypt permissions.

---

Nice find - I'd not seen that article before. Happy to use your suggested change.